pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -353,3 +376,19 @@ func (w *worker) findNodeTemplateAndMachineTypeByPoolName(ctx context.Context, o
 	}
 	return nil, ""
 }
+
+// CheckWorkerStatusMachineDeployments checks if the status of the worker is updated or not during its reconciliation.
+// It is updated if
+// * The status.MachineDeploymentsLastUpdateTime > the value of the time stamp stored in worker struct before the reconciliation begins.
+func (w *worker) CheckWorkerStatusMachineDeployments(o client.Object) error {
```

Comment:
```suggestion
func (w *worker) checkWorkerStatusMachineDeployments(o client.Object) error {
```

Reply:
addressed in [e26191c](https://github.com/gardener/gardener/pull/7774/commits/e26191c2e5ac76a299562a991fe601d380810a5d)